### PR TITLE
Update runahead limit documentation.

### PR DESCRIPTION
**Pull request by hjoliver**
Goes with https://github.com/cylc/cylc-flow/pull/4906

Draft till I read through it myself, fix cross references, and check the build.

### CODE REVIEW

**Comment by hjoliver** [comment]
This one needs to be in 8.0  (https://github.com/cylc/cylc-flow/pull/4906 has been merged).

File: src/user-guide/writing-workflows/scheduling.rst [hunk 1]
```diff
@@ -2013,43 +2011,84 @@ tools to limit workflow activity.
 Runahead Limiting
 ^^^^^^^^^^^^^^^^^
 
-Runahead limiting, controlled by :cylc:conf:`[scheduling]runahead limit`,
-prevents the fastest tasks in a workflow from getting too far
-ahead (with respect to cycle point) of the slowest ones.
+The scheduler runahead limit determines how many consecutive cycle points can
```
Comment (wxtim) [inline src/user-guide/writing-workflows/scheduling.rst:2014]: Do the cycle points have to be consecutive to be affected by the runahead limit?

File: src/user-guide/writing-workflows/scheduling.rst [hunk 2]
```diff
@@ -2013,43 +2011,84 @@ tools to limit workflow activity.
...
+   queues <InternalQueues>` to constrain activity in absolute terms.
+
+
+.. note::
 
-A high runahead limit allows fast tasks that are not constrained by dependence
-on other tasks or clock-triggers to spawn many cycles ahead.
-:ref:`Internal queues <InternalQueues>` may be needed as well, if this would
-result in too many tasks submitting at once.
+   The runahead limit may be raised automatically to accommodate
```
Comment (wxtim) [inline src/user-guide/writing-workflows/scheduling.rst:2089]:
```suggestion
   The scheduler may automatically raise the runahead limit to accommodate
```

File: src/user-guide/writing-workflows/scheduling.rst [hunk 3]
```diff
@@ -2013,43 +2011,84 @@ tools to limit workflow activity.
...
+The scheduler runahead limit determines how many consecutive cycle points can
+be active at once. The base point of the runahead calculation is the current
+lowest-valued point with :term:`active` or :term:`incomplete` tasks present.
 
-.. note::
+The runahead limit can be set with :cylc:conf:`[scheduling]runahead limit`,
+to an integer interval (for datetime or integer cycling workflows) or to
+a datetime interval (datetime cycling only). 
 
-   Runahead limiting does not restrict activity within a cycle point.
```
Comment (wxtim) [inline src/user-guide/writing-workflows/scheduling.rst:2022]: Is this note not worth keeping? I think it helps clarify what a runahead limit can and can't do. Especially now queues actually have a performance benefit?

**Review by wxtim — APPROVED** [review]
Trivial thoughts added. Happy enough.

File: src/user-guide/writing-workflows/scheduling.rst (hunk 3 above)
Comment (hjoliver) [reply src/user-guide/writing-workflows/scheduling.rst:2022]: Yes it is - I just moved it, and highlighted it in an "note" box. See lines 2080-2084 below

File: src/user-guide/writing-workflows/scheduling.rst (hunk 1 above)
Comment (hjoliver) [reply src/user-guide/writing-workflows/scheduling.rst:2014]: Yes, the runahead limit is computed by counting consecutive cycle points. (We've considered using just active cycle points, but don't have that at the moment).

File: src/user-guide/writing-workflows/scheduling.rst (hunk 3 above)
Comment (wxtim) [reply src/user-guide/writing-workflows/scheduling.rst:2022]: Sorry - missed that.

**Pull request merged**